src/libs/selectedAccount/selectedAccount.ts

```diff
@@ -47,80 +45,73 @@ export const updatePortfolioStateWithDefiPositions = (
-              tokens = tokens.filter((_, index) => index !== tokenInPortfolioIndex)
+              networkBalance += protocolTokenBalanceUSD || 0
             }
 
-            // Add only the balance of the collateral tokens to the network balance
-            if (a.type === AssetType.Collateral) {
-              const protocolPriceUSD = a.priceIn.find(
+            if (tokenInPortfolio) {
+              const priceUSD = tokenInPortfolio.priceIn.find(
                 ({ baseCurrency }: { baseCurrency: string }) => baseCurrency.toLowerCase() === 'usd'
               )?.price
 
-              const protocolTokenBalanceUSD = protocolPriceUSD
+              const tokenBalanceUSD = priceUSD
                 ? Number(
                     safeTokenAmountAndNumberMultiplication(
-                      BigInt(amountPostSimulation || a.amount),
-                      Number(a.protocolAsset!.decimals),
-                      protocolPriceUSD
+                      BigInt(tokenInPortfolio.amountPostSimulation || tokenInPortfolio.amount),
+                      tokenInPortfolio.decimals,
+                      priceUSD
                     )
                   )
                 : undefined
 
-              networkBalance += protocolTokenBalanceUSD || 0
-            }
-
-            const positionAsset = {
-              amount: a.amount,
-              amountPostSimulation: a.amountPostSimulation,
-              simulationAmount: a.simulationAmount,
-              // Only list the borrowed asset with no price
-              priceIn: a.type === AssetType.Collateral ? a.priceIn : [],
-              decimals: Number(a.protocolAsset!.decimals),
-              address: a.protocolAsset!.address,
-              symbol: a.protocolAsset!.symbol,
-              networkId,
-              flags: {
-                canTopUpGasTank: false,
-                isFeeToken: false,
-                onGasTank: false,
-                rewardsType: null
+              networkBalance -= tokenBalanceUSD || 0 // deduct portfolio token balance
+              // Get the price from defiPositions
+              tokenInPortfolio.priceIn = a.type === AssetType.Collateral ? a.priceIn : []
+            } else {
+              const positionAsset = {
+                amount: a.amount,
+                // Only list the borrowed asset with no price
+                priceIn: a.type === AssetType.Collateral ? a.priceIn : [],
+                decimals: Number(a.protocolAsset!.decimals),
+                address: a.protocolAsset!.address,
+                symbol: a.protocolAsset!.symbol,
+                networkId,
+                flags: {
+                  canTopUpGasTank: false,
+                  isFeeToken: false,
+                  onGasTank: false,
+                  rewardsType: null
+                  // @BUG: defi positions tokens can't be hidden and can be added as custom
+                  // because processTokens is called in the portfolio
```

Comment:
let's add a link to this issue here: https://github.com/AmbireTech/ambire-app/issues/3971